customer/api.go

```diff
@@ -0,0 +1,178 @@
+package customer
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+type API struct {
+	clientID   string
+	license    int32
+	HTTPClient *http.Client
+	ApiURL     string
+	token      string
+	tokenLock  sync.Mutex
+}
+
+func NewAPI(clientID string, license int32) *API {
+	return &API{
+		clientID:   clientID,
+		license:    license,
+		ApiURL:     "https://api.livechatinc.com/",
+		HTTPClient: http.DefaultClient,
+	}
+}
+
+// SetToken sets customer token. This is required for requests authorization.
+func (a *API) SetToken(t string) {
```

Comment:
I think that instead of setting the token, user should provide function for retrieving the token.